libraries/chain/include/eosio/chain/webassembly/eos-vm-oc/eos-vm-oc.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <setjmp.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+#include <vector>
+#include <list>
+namespace eosio { namespace chain {class apply_context;}}
+#endif
+
+struct eos_vm_oc_control_block {
+   uint64_t magic;
+   uintptr_t execution_thread_code_start;
+   size_t execution_thread_code_length;
+   uintptr_t execution_thread_memory_start;
+   size_t execution_thread_memory_length;
+#ifdef __cplusplus
+   eosio::chain::apply_context* ctx;
+   std::exception_ptr* eptr;
+#else
+   void* ctx;
+   void* eptr;
+#endif
```

Comment:
I think it's better not to have the struct use different types for C vs. C++.  You can add a typed (non-member) accessor if needed.